libs/firebase_config.py

```diff
@@ -0,0 +1,118 @@
+import json
+from json import JSONDecodeError
+
+from firebase_admin import (delete_app as delete_firebase_instance, get_app as get_firebase_app,
+    initialize_app as initialize_firebase_app)
+from firebase_admin.credentials import Certificate as FirebaseCertificate
+
+from config.constants import (ANDROID_FIREBASE_CREDENTIALS, BACKEND_FIREBASE_CREDENTIALS,
+    FIREBASE_APP_TEST_NAME, IOS_FIREBASE_CREDENTIALS)
+from database.system_models import FileAsText
+
+
+class FirebaseMisconfigured(Exception): pass
+
+#
+# Firebase app object instantiation and credential tests
+# (This code can probably be simplified with a threading.Lock object.)
+#
+
+def safely_get_db_credential(credential_type: str) -> str or None:
+    """ just a wrapper to handle ugly code """
+    credentials = FileAsText.objects.filter(tag=credential_type).first()
+    if credentials:
+        return credentials.text
+    else:
+        return None
+
+
+def get_firebase_credential_errors(credentials: str):
+    """ Wrapper to get error strings for test_firebase_credential_errors because otherwise the
+        code is gross.  Returns None if no errors occurred. """
+    try:
+        test_firebase_credential_errors(credentials)
+        return None
+    except Exception as e:
+        return str(e)
+
+
+def test_firebase_credential_errors(credentials: str) -> None:
```

Comment:
Since prefix `test_` is commonly used for test suite functions, rename to something like `check_firebase_credential_errors`?